lib/mongo/server/connection_pool.rb

```diff
@@ -23,6 +23,9 @@ class Server
     class ConnectionPool
       include Loggable
 
+      # @return [ Mutex ] mutex The mutex used for synchronization.
+      attr_reader :mutex
```

Comment:
Doesn't have to be - we could just reference the instance variable directly. Was just following the pattern from the other mutexes.